src/Filters/Traits/AppliesFiltersTrait.php

```diff
@@ -37,15 +37,17 @@ protected function applyFiltersToQuery(array $filters, Builder $query): Builder
 
             // Append time to date, to query with <= and >= like Elasticsearch, instead of using DB::raw('DATE(column)')
             if (strpos($name, 'date_from') !== false || strpos($name, 'date_to') !== false) {
-                if (!preg_match('/^(\d{4}-\d{2}-\d{2})|(\d+)$/', $value)) {
-                    throw new UnprocessableEntityException('The filter ' . $name . ' is not a timestamp or in ISO 8601 date format.');
-                }
-                // We can safely assume that a timestamp is used if it's numeric.
-                if (is_numeric($value)) {
+                // Date string 0000-00-00
+                if (preg_match('/^\d{4}-\d{2}-\d{2}$/', $value)) {
+                    $value .= (strpos($name, 'date_to') !== false) ? ' 23:59:59' : ' 00:00:00';
+                    // ISO 8601 date string 0000-00-00T00:00:00+0000
+                } elseif (preg_match('/^\d{4}-\d{2}-\d{2}T[0-2][0-9]:[0-5][0-9]:[0-5][0-9]\+\d{4}$/', $value)) {
```

Comment:
To fully support ISO 8601 date strings, I think we should also support optional milliseconds`.000` and ending with a `Z` instead of `+0000`.
For example in your browser console: `new Date().toISOString()` gives `2021-12-08T09:33:49.000Z`

Reply:
Oops I merged and then I saw your comment @M4tini . Perhaps there is a public regex that we can use @NickVries ?

Reply:
```
^\d{4}-\d{2}-\d{2}T[0-2][0-9]:[0-5][0-9]:[0-5][0-9](\.\d{3})?(Z|\+\d{4}|\+\d{2}:\d{2})$
```

Reply:
Final version:

`^\d{4}-(?:0[1-9]|1[0-2])-(?:[0-2][0-9]|3[0-1])T(?:[0-1][0-9]|2[0-3]):[0-5][0-9]:[0-5][0-9](\.\d{3})?(Z|\+\d{2}:?\d{2})$`